plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineBundle.kt

```diff
@@ -85,6 +89,61 @@ open class GenerateGramineBundle @Inject constructor(
         }
     }
 
+    private fun findJavaModules(): String {
+        val enclaveJar = outputDir.file(GRAMINE_ENCLAVE_JAR).get().asFile.absolutePath
+
+        val dependentModules = execCommand(
+            "jdeps",
+            "--print-module-deps",
+            "--ignore-missing-deps",
+            enclaveJar
+        ).run { trimEnd().replace(System.lineSeparator(), "").split(",") }
+        // TODO: Provide mechanism to allow the user to add modules needed when using reflection
+        val defaultModules = listOf("java.base", "java.desktop", "java.sql", "jdk.crypto.ec")
+        return (defaultModules + dependentModules).distinct().joinToString(separator = ",")
+    }
+
+    private fun createCustomJDK() {
+        val jlinkOutputPath = "${outputDir.get()}/$JLINK_OUTPUT_DIRECTORY"
+
+        if (Paths.get(jlinkOutputPath).exists()) {
+            File(jlinkOutputPath).deleteRecursively()
+        }
+        val modules = findJavaModules()
+        execCommand(
+            "jlink",
+            "--strip-native-debug-symbols", "objcopy=/usr/bin/objcopy",
+            "--no-header-files",
+            "--no-man-pages",
+            "--add-modules", modules,
+            "--output", jlinkOutputPath
+        )
+    }
+
+    private fun copyAdditionalSystemFiles() {
+        //  These files need to be available when running java inside Gramine (leveraging LD_LIBRARY_PATH)
+        //  together with the OS files which are provided inside Gramine itself.
+        val files = listOf(
+            "/usr/lib/x86_64-linux-gnu/libz.so.1",
+            "/usr/lib/x86_64-linux-gnu/libstdc++.so.6",
+            "/usr/lib/x86_64-linux-gnu/libgcc_s.so.1"
+        )
+        val outputSystemDir = outputDir.get().asFile.toPath() / SYSTEM_LIB_DIRECTORY
+
+        if (outputSystemDir.exists()) {
+            outputSystemDir.toFile().deleteRecursively()
+        }
+        files.forEach {
+            val filePath = Paths.get(outputSystemDir.absolutePathString() + it)
```

Comment:
`outputSystemDir.resolve(it)` should work as well.

Reply:
It gets messed up because of the initial `/`. I would leave it as it is.